core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -1866,16 +2024,56 @@ public static boolean isXMLWellFormed( ValueMetaInterface metaA, Object dataA )
    * @param dataA
    *          The value (filename)
    * @return file encoding.
+   * @deprecated Use {@link ValueDataUtil#getFileEncoding(ValueMetaInterface, Object, boolean)} instead
    */
+  @Deprecated
   public static String getFileEncoding( ValueMetaInterface metaA, Object dataA ) throws KettleValueException {
+    String encoding = null;
+    try {
+      encoding = getFileEncoding( metaA, dataA, true );
+    } catch ( KettleFileNotFoundException e ) {
+      throw new KettleValueException();
+    }
+    return encoding;
+  }
+
+  /**
+   * Get file encoding.
+   *
+   * @param metaA
+   *          The ValueMetaInterface
+   * @param dataA
+   *          The value (filename)
+   * @param failIfNoFile
+   *          Indicates if the transformation should fail if no file is found
+   * @return file encoding.
+   * @throws KettleFileNotFoundException
+   * @throws KettleValueException
+   */
+  public static String getFileEncoding( ValueMetaInterface metaA, Object dataA, boolean failIfNoFile )
+          throws KettleValueException, KettleFileNotFoundException {
     if ( dataA == null ) {
+      if ( failIfNoFile ) {
```

Comment:
This actually is not a file not found.... it's a file path/name not provided.... I'm not sure the file not found exception suits this case